examples/examples/subscribe_runtime_updates.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! To run this example, a local polkadot node should be running. Example verified against polkadot 0.9.18-f6d6ab005d-aarch64-macos.
+//!
+//! E.g.
+//! ```bash
+//! curl "https://github.com/paritytech/polkadot/releases/download/v0.9.13/polkadot" --output /usr/local/bin/polkadot --location
+//! polkadot --dev --tmp
+//! ```
+
+use sp_keyring::AccountKeyring;
+use std::time::Duration;
+use subxt::{
+    ClientBuilder,
+    DefaultConfig,
+    PairSigner,
+    PolkadotExtrinsicParams,
+};
+
+#[subxt::subxt(runtime_metadata_path = "examples/polkadot_metadata.scale")]
+pub mod polkadot {}
+
+type RuntimeApi =
+    polkadot::RuntimeApi<DefaultConfig, PolkadotExtrinsicParams<DefaultConfig>>;
+
+/// This function is representative of the main customer use case.
+async fn user_use_case(api: &RuntimeApi) {
+    let signer = PairSigner::new(AccountKeyring::Alice.pair());
+
+    // Make small balance transfers from Alice to Bob:
+    for _ in 0..10 {
+        let hash = api
+            .tx()
+            .balances()
+            .transfer(
+                AccountKeyring::Bob.to_account_id().into(),
+                123_456_789_012_345,
+            )
+            .sign_and_submit_default(&signer)
+            .await
+            .unwrap();
+
+        println!("Balance transfer extrinsic submitted: {}", hash);
+        tokio::time::sleep(Duration::from_secs(30)).await;
+    }
+}
+
+/// This function handles runtime updates via subscribing to the
+/// node's RuntimeVersion.
+async fn runtime_update(api: &RuntimeApi) {
```

Comment:
I do believe that this should exist by default on the subxt `Client`. I would move this when we handle `Metadata` injection to the `Client`.

What do you think?